src/modules/todo.js

```diff
@@ -0,0 +1,166 @@
+        todoCheck.checked = true;
+        todoInput.style.textDecoration = 'line-through';
+        todoInput.style.color = '#4169e1';
+      } else {
+        todoCheck.checked = false;
+        todoInput.style.textDecoration = '';
+        todoInput.style.color = '';
+      }
+      todoLi.append(todoCheck, todoInput, editElement, removeElement);
+      todoContainer.appendChild(todoLi);
+    });
+  }
+
+  static deleteTodo() {
+    const del = document.querySelectorAll('.remove-icon');
+    const todoContainer = document.getElementById('todo-list');
+    del.forEach((delBtn, index) => {
+      delBtn.addEventListener('click', () => {
+        const todoList = JSON.parse(localStorage.getItem('todoList') || '[]');
+        todoList.splice(index, 1);
+        todoContainer.remove();
+        todoList.forEach((todo, i) => {
+          todo.index = i + 1;
+        });
+        localStorage.setItem('todoList', JSON.stringify(todoList));
+        window.location.reload();
+      });
+    });
+  }
+
+  static editTodo() {
+    const todoItems = document.querySelectorAll('.todo-item');
+    const editIcons = document.querySelectorAll('.edit-icon');
+    editIcons.forEach((edit) => {
+      edit.addEventListener('click', (e) => {
+        todoItems.forEach((item) => {
+          if (e.target.id === item.id) item.removeAttribute('readonly');
+        });
+      });
+      todoItems.forEach((todo) => {
+        todo.addEventListener('focusout', () => {
+          const todoList = JSON.parse(localStorage.getItem('todoList') || '[]');
+          todoList.forEach((todoli) => {
+            if (todoli.index.toString() === todo.id) {
+              todoli.description = todo.value;
+              localStorage.setItem('todoList', JSON.stringify(todoList));
+              todo.setAttribute('readonly', true);
+            }
+          });
+        });
+      });
+    });
+  }
+
+  static isCompleted() {
+    const checked = document.querySelectorAll('.select');
+    const todoItems = document.querySelectorAll('.todo-item');
+
+    checked.forEach((itemCheck) => {
+      itemCheck.addEventListener('change', (e) => {
+        const todoList = JSON.parse(localStorage.getItem('todoList') || '[]');
+
+        todoList.forEach((item) => {
+          if ((item.completed === false) && (+e.target.id === item.index)) {
+            item.completed = true;
+            todoItems.forEach((todoI) => {
+              if (todoI.id === item.index.toString() && item.completed) { todoI.classList.add('checked'); }
+            });
+            localStorage.setItem('todoList', JSON.stringify(todoList));
+          } else if ((item.completed === true) && (+e.target.id === item.index)) {
+            item.completed = false;
+            todoItems.forEach((todoI) => {
+              if (todoI.id === item.index.toString() && !item.completed) { todoI.classList.remove('checked'); }
+            });
+            localStorage.setItem('todoList', JSON.stringify(todoList));
+          }
```

Comment:
- [ ] Please implement this is a new js file and import it here as a module to use it as demanded by the [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#project-requirements).